[CL-712] Update icon button, components using it, and affected virtual scroll heights

vleague2 opened the pull request:
## 🎟️ Tracking

<!-- Paste the link to the Jira or GitHub issue or otherwise describe / point to where this change is coming from. -->
[CL-712](https://bitwarden.atlassian.net/browse/CL-712)

## 📔 Objective

<!-- Describe what the purpose of this PR is, for example what bug you're fixing or new feature you're adding. -->
This PR makes updates to the icon button component, as well as adjustments to some components that contain icon buttons. Virtual scroll heights are also updated to account for the sizing updates to the icon button.

## 📸 Screenshots

<!-- Required for any UI changes; delete if not applicable. Use fixed width images for better display. -->
Too many to screenshot. Check the following stories:
- Icon Button
- Dialog and Drawer
- Banner
- Item
- Popup Layout
- Form Field
- Search
- Popover
- Nav
- Spotlight
- Table
- Toast

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

<!-- Suggested interactions but feel free to use (or not) as you desire! -->

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes


[CL-712]: https://bitwarden.atlassian.net/browse/CL-712?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

emily-livefront commented:
Hi @vleague2! A few visual refinement notes from me below. Let me know if its helpful to chat more or jump on a quick huddle about anything.

**Icon button**
- Forgot to call this out specifically in the updates, but the Icon button-small corner radius should be changed to 4px instead of 8px
- [LINK] (https://www.figma.com/design/Zt3YSeb6E6lebAffrNLa0h/Tailwind-Component-Library?node-id=2133-17681&t=YRD4xNo0hZprwnqs-11)


**Drawer**
- Missing 16px spacing between header elements (leading icon, header text, and icon button)
- [LINK] (https://www.figma.com/design/Zt3YSeb6E6lebAffrNLa0h/Tailwind-Component-Library?node-id=21753-19037&t=YRD4xNo0hZprwnqs-11)
<img width="374" height="74" alt="image" src="https://github.com/user-attachments/assets/9524e1dc-1fe5-4304-b9fd-6af986c86cf2" />


**Header/Extension**
- It looks like in the implementation the account switcher button has extra L/R padding that we don’t have in the design, which is making the header element spacing look uneven
- Could we remove the L/R padding from this element?
- [LINK](https://www.figma.com/design/Zt3YSeb6E6lebAffrNLa0h/branch/s5XkKAid2KARdP9ik5MyWH/Tailwind-Component-Library?node-id=12921-127194&t=uArO1AX7FPONmLeE-11)
<img width="391" height="74" alt="Pasted Graphic" src="https://github.com/user-attachments/assets/c730febd-90e2-487e-a810-830c5eaca773" />


**Item**
- Add 8px padding on the right side of the “Fill” chip button
- [LINK](https://www.figma.com/design/Zt3YSeb6E6lebAffrNLa0h/branch/s5XkKAid2KARdP9ik5MyWH/Tailwind-Component-Library?node-id=12921-128246&t=uArO1AX7FPONmLeE-11)
<img width="155" height="112" alt="Pasted Graphic 1" src="https://github.com/user-attachments/assets/a9c2ab5d-1d47-4ed3-850d-c53bd9808bcc" />


**Popover**
- The text should appear vertically centered with the “Close” icon button, can we add 4px spacing above the title text so the top line of text will always look vertically centered with the button?
- [LINK](https://www.figma.com/design/s5XkKAid2KARdP9ik5MyWH/Icon-button-hover-state-updates?node-id=9768-33761&t=uArO1AX7FPONmLeE-4)
<img width="358" height="105" alt="Example Title" src="https://github.com/user-attachments/assets/2ce7158b-6c2f-4447-a4fd-e7aac9079418" />

vleague2 commented:
@emily-livefront 

**Icon button**
- Fixed, thanks for catching!

**Drawer**
- The drawer in storybook is deprecated because we have the dialogs opening as drawers now. I updated the deprecated drawer anyway, but can you check the Dialog > Service > Drawer story to make sure that looks as expected since that is the one that will be used moving forward?

**Header/extension**
- The account switcher in storybook is not the real one, just a placeholder, so it's not accurate for what the account switcher looks like in the actual extension. The real one looks ok as far as padding. I updated the storybook placeholder one so that it looks closer to the one in the extension

**Item**
- Let me know if this is acceptable as far as the gap between the elements. If not, then the teams (I think vault is the only one that does this implementation) will need to be responsible for adding the right padding to the “fill” button, because we allow anything to be put in that “actions on the righthand side” slot and can’t specifically know that a badge button needs right padding in a specific circumstance
<img width="380" height="325" alt="Screenshot 2025-07-21 at 3 43 12 PM" src="https://github.com/user-attachments/assets/6b3174d0-169e-4b0d-80b6-f20a758991d0" />

**Popover**
- Updated

danielleflinn commented:
Just re-reviewed the Chromatic difs – I have an open question to the design team on if we want to adjust the navigation links as the right spacing on the icon button looks off, but we have this in Figma so it might be alright:
<img width="154" height="86" alt="Screenshot 2025-07-22 at 4 59 37 PM" src="https://github.com/user-attachments/assets/15839d75-ea4d-47bb-bf11-7da0eb76edd5" />

I did catch a few things we might want to address in team owned code:
- the [Spotlight component](https://62a88a6de5b807fa98886113-nyiwsyyzvi.chromatic.com/?path=/story/vault-spotlight--default) close button should probably be visually aligned similar to how we did in other components:
   - Reduce top and right padding to 8px
   - Add 4px of right padding to button so it lines up with the close option
   - [Figma visual](https://www.figma.com/design/Zt3YSeb6E6lebAffrNLa0h/Tailwind-Component-Library?node-id=28947-37484&t=8jqyoYhwwlnabtpC-4)

- the Vault Website/URI cog has some odd right padding that is more noticeable after this change
 <img width="631" height="196" alt="Screenshot 2025-07-22 at 4 38 59 PM" src="https://github.com/user-attachments/assets/87087bf4-2c25-4a88-9442-645d37257dd1" />

@vleague2 let me know what you think about these team code visual tweaks as I know that wasn't initially in the scope of this jira. 

@emily-livefront let me know when you've had a chance to look over the updates Vicki made in response to your feedback.

danielleflinn reviewed (dismissed):
✅ Chromatic UI Review approved by design
✅ Chromatic UI Diffs reviewed by design
ℹ️ Dialog header does not match Figma but Bryan's changes in https://github.com/bitwarden/clients/pull/15660 should resolve

Thanks Vicki for all your effort on this one

BryanCunningham reviewed (dismissed):
👍 Seems good to me
